org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTTest.java

```diff
@@ -1312,46 +1309,31 @@ void genericPropertyListTest(ASTNode node, List children, Property prop) {
 
 	@SuppressWarnings("deprecation")
 	private static int getApiLevel(String s) {
-		if (s == null)
-			return AST.JLS12;
-		switch (s) {
-		case JavaCore.VERSION_1_2 : return AST.JLS2;
-        case JavaCore.VERSION_1_3: return AST.JLS3;
-        case JavaCore.VERSION_1_4: return AST.JLS4;
-        case JavaCore.VERSION_1_5: return AST.JLS4;
-        case JavaCore.VERSION_1_6: return AST.JLS4;
-        case JavaCore.VERSION_1_7: return AST.JLS4;
-        case JavaCore.VERSION_1_8: return AST.JLS8;
-        case JavaCore.VERSION_9: return AST.JLS9;
-        case JavaCore.VERSION_10: return AST.JLS10;
-        case JavaCore.VERSION_11: return AST.JLS11;
-        case JavaCore.VERSION_12: return AST.JLS12;
-        case JavaCore.VERSION_13: return AST.JLS13;
-        case JavaCore.VERSION_14: return AST.JLS14;
-        case JavaCore.VERSION_15: return AST.JLS15;
-        case JavaCore.VERSION_16: return AST.JLS16;
-        case JavaCore.VERSION_17: return AST.JLS17;
-        case JavaCore.VERSION_18: return AST.JLS18;
-        case JavaCore.VERSION_19: return AST.JLS19;
-        case JavaCore.VERSION_20: return AST.JLS20;
-        default:  return AST.JLS2;
-		}
+        try {
+            int apiLevel = Integer.valueOf(s);
+            if (AST.isSupportedVersion(apiLevel)) {
+                return apiLevel;
+            }
+        } catch (NumberFormatException e) {
+            if (s.equals(JavaCore.VERSION_1_8)) {
```

Comment:
Wouldn't be better to have a dedicated `equals(JavaCore.VERSION_1_8)` check before and hard fail on anything that throws NumberFormatException? Or are there still too many 1.x users in the tests?

Reply:
Done.